README.md

```diff
@@ -1,2 +1,17 @@
-# comp110-worksheet-C
-Base repository for COMP110 worksheet C
+a) It checks if a list contains a duplicate.
+
+b) Because if the list contains no duplicates, it will have to run through the second again for every additional element.
+
+c) Because the second duplicate item's iteration will  always catch the fact that it matched the first.
+
+d) Because on average, it will only go through just under two-thirds of the iterations (1+2+3...+n instead of n^2).
```

Comment:
Why two thirds? 1+2+...+n = 1/2 * n * (n-1) ~ 1/2 * n^2, so it will have on average half the iterations.